### ISSUE TITLE
Hash#merge (with no argument) should return a new Hash

### ISSUE THREAD

**Issue by Mogztter** [oracle/truffleruby#1645]
### Code

**hash-merge.rb**
```rb
a = {'foo' => 'bar'}
b = a.merge
a.clear
p b
```

### MRI

```
$ ruby --version
```
```
ruby 2.6.0p0 (2018-12-25 revision 66547) [x86_64-linux]
```
```
$ ruby hash-merge.rb   
{"foo"=>"bar"}
```

### TruffleRuby 1.0.0-rc.15

```
$ ruby --version
```
```
truffleruby 1.0.0-rc15, like ruby 2.6.2, GraalVM CE Native [x86_64-linux]
```
```
$ ruby hash-merge.rb   
{}
```

It looks like TruffleRuby does not create a new Hash.

**hash-id.rb**
```rb
a = {'foo' => 'bar'}
b = a.merge
p a.__id__
p b.__id__
```

With MRI we can see that the `__id__` of `a` and `b` are different:

```
$ ruby
47022355548920
47022355548900
```

But with TruffleRuby, the `__id__` are identical:

```
$ ruby
520
520
```

**Comment by nirvdrum** [comment]
Thanks for the detailed report! I've fixed the issue in f5b069ce797f95db9d26ccdb4470afeb14d71c71 and updated a spec to represent the correct behavior. This should appear in the next release (1.0.0-rc16).

**Comment by Mogztter** [comment]
Awesome thanks @nirvdrum
